### PR TITLE
Doc Blocks: Add support for `of` prop to `Stories` block

### PR DESCRIPTION
**Pull request by jorge-ji**
Co-authored-by: @jiyiru 

Partially implements #22490

## What I did

- Add support for `of` prop via useOf with "meta" as only valid type
- Added stories
- Default to get title from parameters.docs.stories.title
- Default to get includePrimaryStory from parameters.docs.stories.includePrimaryStory
- Updated documentation, using https://github.com/storybookjs/storybook/pull/22552 as an example

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run yarn storybook:blocks
2. Open Storybook in your browser
3. Access Stories stories

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/ui/blocks/src/blocks/Stories.tsx
Comment (shilman) [inline code/ui/blocks/src/blocks/Stories.tsx:12]: This name change is breaking, so it can't be included in a minor release. Please revert or we will need to wait until Storybook 9 to merge this PR.

**Review by shilman — CHANGES_REQUESTED** [review]
Thank you @jorge-ji and @jiyiru ! 🙏  Needs a minor change to be considered for merging. Please let me know if you have any questions

File: code/ui/blocks/src/blocks/Stories.tsx
Comment (jorge-ji) [reply code/ui/blocks/src/blocks/Stories.tsx]: Thanks for reviewing the code! I reverted includePrimaryStories back to includePrimary.

**Pull request closed without merging**
